cdap-app-fabric/src/test/java/io/cdap/cdap/internal/app/runtime/RemoteAppStateStoreTest.java

```diff
@@ -98,95 +179,88 @@ public static void cleanUp() throws Exception {
 
   @Test
   public void testSaveSuccess() throws IOException {
-    RemoteAppStateStore remoteAppStateStore = new RemoteAppStateStore(cConf, remoteClientFactory, NAMESPACE, "app1");
-    byte[] value = "testvalue".getBytes();
-    remoteAppStateStore.saveState("key1", value);
+    RemoteAppStateStore remoteAppStateStore = new RemoteAppStateStore(cConf, remoteClientFactory,
+                                                                      NAMESPACE, SUCCESS_APP);
+    remoteAppStateStore.saveState("valid-key", "some_value".getBytes(StandardCharsets.UTF_8));
+  }
+
+  @Test
+  public void testSaveSuccessWithSpaceInKey() throws IOException {
+    RemoteAppStateStore remoteAppStateStore = new RemoteAppStateStore(cConf, remoteClientFactory,
+                                                                      NAMESPACE, SUCCESS_APP);
+    remoteAppStateStore.saveState("valid key with space", "some_value".getBytes(StandardCharsets.UTF_8));
   }
 
   @Test
   public void testSaveInvalidNamespace() throws IOException {
     expectedException.expectCause(CoreMatchers.isA(NotFoundException.class));
-    RemoteAppStateStore remoteAppStateStore = new RemoteAppStateStore(cConf, remoteClientFactory, "invalid", "app1");
-    byte[] value = "testvalue".getBytes();
-    remoteAppStateStore.saveState("key1", value);
+    RemoteAppStateStore remoteAppStateStore = new RemoteAppStateStore(cConf, remoteClientFactory,
+                                                                      "invalid", "some_app");
```

Comment:
nit: indentation

Reply:
Indentation matches with the method params.